rig/machine_control/machine_controller.py

```diff
@@ -2110,6 +2132,19 @@ def chips(self):
         """
         return iter(self)
 
+    def ethernet_connected_chips(self):
+        """Iterate over the coordinates of Ethernet connected chips.
+
+        Yields
+        ------
+        ((x, y), str)
+            The coordinate and IP address of each Ethernet connected chip in
+            the system.
+        """
+        return ((xy, chip_info.ip_address)
```

Comment:
Yeah, sorry, I did mean `chip_info.ip_address` - I mostly meant that I felt explicit use of the `yield` keyword would be better.